cpp/src/slice2java/Gen.cpp

```diff
@@ -136,6 +136,28 @@ string ofFactory(const TypePtr& type)
     return "java.util.Optional.ofNullable";
 }
 
+// Returns the "length" of the constructor parameter list from the given data member list.
+// All types are counted as 1, except for long and double which are counted as 2.
+// See https://docs.oracle.com/javase/specs/jvms/se20/html/jvms-4.html#jvms-4.3.3
+int
+constructorParameterLength(const DataMemberList& members)
+{
+    int length = 0;
+    for(DataMemberList::const_iterator i = members.begin(); i != members.end(); ++i)
```

Comment:
I think `p` is a more common iterator name.

Reply:
I think both are pretty common, I can change to `p` though.